src/blocks/weather.rs

```diff
@@ -331,6 +392,34 @@ fn convert_wind_direction(direction_opt: Option<f64>) -> &'static str {
     }
 }
 
+// Compute the average wind speed and direction
+fn average_wind(winds: &[Wind]) -> Wind {
+    let mut north = 0.0;
+    let mut east = 0.0;
+    let mut count = 0.0;
+    for wind in winds {
+        if let Some(degrees) = wind.degrees {
+            let (sin, cos) = degrees.to_radians().sin_cos();
+            north += wind.speed * sin;
+            east += wind.speed * cos;
+            count += 1.0;
+        }
+    }
+    if count == 0.0 {
+        Wind {
+            speed: 0.0,
+            degrees: None,
+        }
+    } else {
+        north /= count;
+        east /= count;
+        Wind {
+            speed: north.hypot(east),
```

Comment:
Oh we can remove a division, by just dividing by the hypotenuse by the count insead of the north and south components.

```math
\displaylines{
\sqrt{(xa)^2+(xb)^2)} = xc^2\\
\sqrt{x^2(a^2+b^2)} = xc^2\\
x\sqrt{a^2+b^2} = xc^2\\
\sqrt{a^2+b^2} = c^2
}
```